### PR TITLE
CXP-1182: Creation of the ServiceAPI

### PR DESCRIPTION
**Pull request by willy-ahva**
**Description (for Contributor and Core Developer)**

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: components/catalogs/back/src/Infrastructure/Messenger/QueryBus.php [hunk 1]
```diff
@@ -18,14 +18,14 @@ final class QueryBus
...
-        MessageBusInterface $messageBus
+        MessageBusInterface $messageBus,
     ) {
         $this->messageBus = $messageBus;
     }
 
     /**
      * @template R
-     * @param Query<R> $query
+     * @param \Akeneo\Catalogs\ServiceAPI\Query\Query<R> $query
```
Comment (RiffVentura) [inline components/catalogs/back/src/Infrastructure/Messenger/QueryBus.php:28]: Should be in a use statement

**Review by RiffVentura — APPROVED** [review]

File: components/catalogs/Makefile [hunk 2]
```diff
@@ -15,6 +15,7 @@ catalogs-tests:
 	$(MAKE) catalogs-lint-back
 	$(MAKE) catalogs-coupling-back
 	$(MAKE) catalogs-coverage-back
+	$(MAKE) catalogs-integration-back
```
Comment (tseho) [inline components/catalogs/Makefile:18]:
```suggestion
	$(MAKE) catalogs-coverage-back
```

catalogs-coverage-back runs both unit & integration tests with coverage enabled, then merge both coverage and you can check the result by opening coverage/Catalogs/Back/Global/index.html in your browser.

**Review by tseho — COMMENTED** [review]
We need to also move both QueryBus & CommandBus to the ServiceAPI namespace

**Review by guillaumefer-akeneo — APPROVED** [review]

File: components/catalogs/back/src/ServiceAPI/Query/GetCatalogQuery.php [hunk 3]
```diff
@@ -2,18 +2,18 @@
...
-use Akeneo\Catalogs\Domain\Model\Catalog;
+use Akeneo\Catalogs\ServiceAPI\Model\Catalog;
 use Symfony\Component\Validator\Constraints as Assert;
 
 /**
  * @copyright 2022 Akeneo SAS (http://www.akeneo.com)
  * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
  *
- * @implements Query<Catalog>
+ * @implements QueryInterface<Catalog>
```
Comment (tseho) [inline components/catalogs/back/src/ServiceAPI/Query/GetCatalogQuery.php:14]:
```suggestion
 * @implements QueryInterface<Catalog|null>
```

**Comment by tseho** [comment]
The CI is green, we finished it during the mob programming yesterday, I'm merging it in `CXP-1183` for the final review.

**Comment by tseho** [comment]
Thanks Willy for the great work on the ServiceAPI !

**Pull request merged**